lms/djangoapps/program_enrollments/models.py

```diff
@@ -51,6 +54,23 @@ class Meta(object):
     status = models.CharField(max_length=9, choices=STATUSES)
     historical_records = HistoricalRecords()
 
+    def clean(self):
+        if not (self.user or self.external_user_key):
+            raise ValidationError(_('One of user or external_user_key must not be null.'))
+
+    @classmethod
+    def bulk_read_by_student_key(cls, program_uuid, student_data):
```

Comment:
Why do we need this class method to read it like this? Is it for testing only?

Reply:
This is something Rick needs for the POST function, but I should tear it out of this PR.